File: samples/C#/WebcamDemo/WebcamDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Dynamsoft.DotNet.TWAIN.Enums;

namespace WebcamDemo
{
    public partial class Form1 : Form
    {
        private int m_iDesignWidth = 755;
        public Form1()
        {
            InitializeComponent();

            dynamicDotNetTwain1.SupportedDeviceType = EnumSupportedDeviceType.SDT_WEBCAM;

            this.chkContainer.Checked = true;

            this.Load += new EventHandler(Form1_Load);
        }

        void Form1_Load(object sender, EventArgs e)
        {
            try
            {
                m_iDesignWidth = this.Width;
                this.chkContainer.CheckedChanged += new System.EventHandler(this.chkContainer_CheckedChanged);
                this.chkContainer.Checked = false;

                for (short i = 0; i < dynamicDotNetTwain1.SourceCount; i++)
                {
                    string strSourceName = dynamicDotNetTwain1.SourceNameItems(i);
                    if (strSourceName != null)
                        cbxSources.Items.Add(strSourceName);
                }

                if (cbxSources.Items.Count > 0)
                    cbxSources.SelectedIndex = 0;
            }
            catch (Exception exp)
            {
                MessageBox.Show(exp.Message);
            }
        }       

        private void btnRemoveAllImages_Click(object sender, EventArgs e)
        {
            dynamicDotNetTwain1.RemoveAllImages();
        }

        private void btnCaptureImage_Click(object sender, EventArgs e)
        {
            try
            {
                if (cbxSources.Items.Count > 0)
                {
                    dynamicDotNetTwain1.IfShowUI = true;
                    if (chkContainer.Checked)
                    {
                        Dynamsoft.DotNet.TWAIN.WebCamera.CamResolution camResolution = dynamicDotNetTwain1.ResolutionForCam;
                        if (camResolution != null && camResolution.Width > 0 && camResolution.Height > 0)
                        {
                            pictureBox1.Height = pictureBox1.Width * camResolution.Height / camResolution.Width; 
                            int iContentHeight = panel1.Height - panel1.Margin.Top - panel1.Margin.Bottom - panel1.Padding.Top - panel1.Padding.Bottom;
                            if (pictureBox1.Height < iContentHeight)
                                pictureBox1.Location = new Point (pictureBox1.Location.X, (iContentHeight - pictureBox1.Height) / 2);
                            else
                                pictureBox1.Location =  new Point (pictureBox1.Location.X, 0);
                        }
                        dynamicDotNetTwain1.SetVideoContainer(this.pictureBox1);
                    }
                    else
                        dynamicDotNetTwain1.SetVideoContainer(null);
                    dynamicDotNetTwain1.EnableSource();
                }
                else
                {
                    MessageBox.Show("No webcam source detected!");
                }
            }
            catch (Exception exp)
            {
                MessageBox.Show(exp.Message);
            }
            
        }

        private void cbxSources_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (((ComboBox)sender).SelectedIndex >= 0 && ((ComboBox)sender).SelectedIndex < dynamicDotNetTwain1.SourceCount)
            {
                dynamicDotNetTwain1.SelectSourceByIndex(cbxSources.SelectedIndex);
                dynamicDotNetTwain1.OpenSource();
            }
        }

        private void chkContainer_CheckedChanged(object sender, EventArgs e)
        {
            if (!chkContainer.Checked)
            {
                lbContainer.Visible = false;
                panel1.Visible = false;
                this.Width = m_iDesignWidth - this.panel1.Width - 15;
            }
            else
            {
                lbContainer.Visible = true;
                panel1.Visible = true;
                this.Width = m_iDesignWidth;
            }
        }
    }
}

File: samples/VB.NET/WebcamDemo/WebcamDemo/Form1.vb
Imports Dynamsoft.DotNet.TWAIN.Enums
Public Class Form1

    Private designWidth As Int32

    Sub New()
        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        dynamicDotNetTwain1.SupportedDeviceType = EnumSupportedDeviceType.SDT_WEBCAM
        Me.chkContainer.Checked = True

    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As EventArgs) Handles Me.Load
        Try
            designWidth = Me.Width
            AddHandler chkContainer.CheckedChanged, AddressOf chkContainer_CheckedChanged
            chkContainer.Checked = False

            Dim i As Int16 : i = 0
            For i = 0 To dynamicDotNetTwain1.SourceCount - 1 Step 1
                Dim strSourceName As String : strSourceName = dynamicDotNetTwain1.SourceNameItems(i)
                If strSourceName IsNot Nothing Then
                    cbxSources.Items.Add(strSourceName)
                End If
            Next i

            If cbxSources.Items.Count > 0 Then
                cbxSources.SelectedIndex = 0
            End If
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub btnRemoveAllImages_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnRemoveAllImages.Click
        dynamicDotNetTwain1.RemoveAllImages()
    End Sub

    Private Sub cbxSources_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbxSources.SelectedIndexChanged
        If (cbxSources.SelectedIndex >= 0 And cbxSources.SelectedIndex < dynamicDotNetTwain1.SourceCount) Then
            dynamicDotNetTwain1.SelectSourceByIndex(cbxSources.SelectedIndex)
            dynamicDotNetTwain1.OpenSource()
        End If
    End Sub

    Private Sub btnCaptureImage_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCaptureImage.Click
        Try

            If dynamicDotNetTwain1.SourceCount > 0 Then
                dynamicDotNetTwain1.IfShowUI = True
                If chkContainer.Checked Then
                    Dim camResolution As Dynamsoft.DotNet.TWAIN.WebCamera.CamResolution : camResolution = dynamicDotNetTwain1.ResolutionForCam
                    If camResolution IsNot Nothing And camResolution.Width > 0 And camResolution.Height > 0 Then
                        pictureBox1.Height = pictureBox1.Width * camResolution.Height / camResolution.Width
                        Dim iContentHeight As Integer : iContentHeight = Panel1.Height - Panel1.Margin.Top - Panel1.Margin.Bottom - Panel1.Padding.Top - Panel1.Padding.Bottom
                        If pictureBox1.Height < iContentHeight Then
                            pictureBox1.Location = New Point(pictureBox1.Location.X, (iContentHeight - pictureBox1.Height) / 2)
                        Else
                            pictureBox1.Location = New Point(pictureBox1.Location.X, 0)
                        End If
                    End If

                    dynamicDotNetTwain1.SetVideoContainer(Me.pictureBox1)
                Else
                    dynamicDotNetTwain1.SetVideoContainer(Nothing)
                End If

                dynamicDotNetTwain1.EnableSource()
            Else
                MessageBox.Show("No webcam source detected!")
            End If
        Catch exp As Exception
            MessageBox.Show(exp.Message)
        End Try
    End Sub

    'Private Sub chkIfShowUI_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs)
    '    chkContainer.Enabled = chkIfShowUI.Checked
    'End Sub

    Private Sub chkContainer_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs)
        If Not chkContainer.Checked Then
            lbContainer.Visible = False
            Panel1.Visible = False
            Me.Width = designWidth - Me.Panel1.Width - 15
        Else
            lbContainer.Visible = True
            Panel1.Visible = True
            Me.Width = designWidth 'Me.Width + Me.pictureBox1.Width + 20
        End If
    End Sub
End Class
